spec/lib/bank_accounts/checking_account_spec.rb

```diff
@@ -0,0 +1,125 @@
+require './spec/spec_helper'
+require './lib/bank_accounts/checking_account'
+
+describe BankAccounts::CheckingAccount do
+
+
+  describe "class methods" do
+
+    context "when balance would be negative" do
+      it "throws an error and aborts savings account creation" do
+        expect { BankAccounts::CheckingAccount.new(1, -1) }.to raise_error(ArgumentError, "Checking account can't have a negative balance. Create a new account.")
+      end
+    end
+
+  end # class methods
+
+
+  describe "attributes" do
+    let(:checking_account) { BankAccounts::CheckingAccount.new(1, 2) }
+
+    it "has initial balance" do
+      expect(checking_account.balance).to eq(2)
+    end
+
+    it "has id" do
+      expect(checking_account.id).to eq(1)
+    end
+
+    it "has blank checks" do
+      expect(checking_account.checks_used).to eq(0)
+    end
+  end # attributes
+
+
+  describe "instance methods" do
+    let(:checking_account) { BankAccounts::CheckingAccount.new(1, 20) }
+
+    it "deposits money" do
+          expect(checking_account.deposit(8)).to eq(28)
+      end
+
+    it "won't accept negative deposits" do
+      expect(checking_account.deposit(-10)).to eq(30)
+    end
+
+    it "withdraws money with fee" do
+      expect(checking_account.withdraw(19)).to eq(0)
+    end
+
+    it "won't accept negative withdrawls with fee" do
+      expect(checking_account.withdraw(-19)).to eq(0)
+    end
+
+    it "won't withdraw too much money with fee" do
+      expect(checking_account.withdraw(20)).to eq(20)
+    end
+
+    it "withdraws money with checks" do
+      expect(checking_account.withdraw_using_check(30)).to eq(-10)
+    end
+
+    it "won't withdraw too much money with checks" do
+      expect(checking_account.withdraw_using_check(31)).to eq(20)
+    end
+
+
+    context "when doing many check withdrawls" do
+      let(:checks_used) {0}
+      before(:each) do
+				checking_account.withdraw_using_check(2)
+			end
+
+      it "counts check usage" do
+        expect(checking_account.checks_used).to eq(1)
+      end
+
+
+      context "when on 3rd check withdrawl" do
+        before(:each) do
```

Comment:
I'm not sure what happened with the formatting in these files - indentation beware!